plugins/action_update_review_status/src/updateReviewsStatuses.ts

```diff
@@ -0,0 +1,184 @@
+
+const updateReviewsStatuses: ActionPluginType = async ({
+  parameters,
+  applicationData,
+  DBConnect,
+}) => {
+  const db = databaseMethods(DBConnect)
+
+  // Get application/reviewId from applicationData if not provided in parameters
+  const applicationId = parameters?.applicationId ?? applicationData?.applicationId
+  const reviewId = parameters?.reviewId ?? applicationData?.reviewData?.reviewId
+  const stageNumber = parameters?.stageNumber || applicationData?.stageNumber
+
+  // Changed responses normally come from the output of the "trimResponses"
+  // action
+  const changedResponses: ChangedResponse[] = parameters.changedResponses || []
+
+  const triggeredBy: TriggeredBy = parameters.triggeredBy || 'APPLICATION'
+
+  console.log(
+    `Updating statuses of reviews associated with ${
+      triggeredBy === 'REVIEW' ? 'review Id: ' + reviewId : 'application Id: ' + applicationId
+    }`
+  )
+
+  try {
+    // Get all reviews for this application/stage. We do additional filtering
+    // (e.g for review level) as required further on.
+    const reviews = await db.getAssociatedReviews(applicationId, stageNumber)
+
+    // APPLICATION SUBMISSIONS:
+    if (triggeredBy === 'APPLICATION') {
+      // Get section codes of the changed responses
+      const changedSections = await db.getChangedSections(
+        changedResponses.map(({ templateElementId }) => templateElementId)
+      )
+
+      // Filter out reviews that are not level 1 and don't intersect with
+      // changed section codes
+      const reviewsToUpdate = reviews.filter(
+        ({ reviewStatus, levelNumber, assignedSections }) =>
+          levelNumber === 1 &&
+          reviewStatus === ReviewStatus.Submitted &&
+          assignedSections.some((sectionCode) => changedSections.includes(sectionCode))
+      )
+
+      // Set remaining reviews to "PENDING" (using the "changeStatus" action)
+      const results: Promise<ActionPluginOutput>[] = []
+      reviewsToUpdate.forEach((review) =>
+        results.push(
+          changeStatus({
+            parameters: {
+              newStatus: ReviewStatus.Pending,
+              reviewId: review.reviewId,
+              isReview: true,
+            },
+            applicationData,
+            DBConnect,
+          })
+        )
+      )
+
+      const reviewStatuses = await Promise.all(results)
+
+      return {
+        status: ActionQueueStatus.Success,
+        error_log: '',
+        output: {
+          updatedReviews: reviewStatuses.map(({ output }) => output),
+        },
+      }
+    } else {
+      // REVIEW SUBMISSIONS:
+      if (!reviewId) throw new Error('Missing reviewId')
+
+      const thisReviewLevel = reviews.find((review) => review.reviewId === reviewId)?.levelNumber
+
+      if (!thisReviewLevel) throw new Error('Invalid reviewId')
+
+      const nextLevelReview = reviews.filter(
+        ({ levelNumber }) => levelNumber === thisReviewLevel + 1
+      )
```

Comment:
Maybe check for not DISCONTINUED Reviews here. In case there was another Review of level +1 which isn't the current ASSIGNED one.

Reply:
That's to cover the case for Re-assignment on consolidations

Reply:
Cool, I've added an additional filter to remove all "Discontinued" reviews right at the top (with comment and not in docs): 6c8b5b22